src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiProductController.cs

```diff
@@ -4,63 +4,61 @@
 using System.Web.Http;
 using Ucommerce.Api;
 using UCommerce.Catalog.Models;
+using UCommerce.EntitiesV2;
+using UCommerce.Extensions;
 using UCommerce.Infrastructure;
 using UCommerce.RazorStore.Api.Model;
 using UCommerce.Search;
-using UCommerce.Search.Models;
-using ProductProperty = UCommerce.RazorStore.Api.Model.ProductProperty;
+using Category = UCommerce.Search.Models.Category;
+using Product = UCommerce.Search.Models.Product;
+using ProductCatalog = UCommerce.Search.Models.ProductCatalog;
 
 namespace UCommerce.RazorStore.Api
 {
     [RoutePrefix("ucommerceapi")]
     public class AvenueClothingApiProductController : ApiController
     {
-        private readonly ICatalogLibrary _catalogLibrary = ObjectFactory.Instance.Resolve<ICatalogLibrary>();
-        private readonly IUrlService _urlService = ObjectFactory.Instance.Resolve<IUrlService>();
-
+        public IUrlService UrlService => ObjectFactory.Instance.Resolve<IUrlService>();
+        public ICatalogLibrary CatalogLibrary => ObjectFactory.Instance.Resolve<ICatalogLibrary>();
+        public ICatalogContext CatalogContext => ObjectFactory.Instance.Resolve<ICatalogContext>();
+        public IIndex<Product> ProductsIndex => ObjectFactory.Instance.Resolve<IIndex<Product>>();
 
         [Route("razorstore/products/getproductvariations")]
         [HttpPost]
         public IHttpActionResult GetProductVariations([FromBody] GetProductVariationsRequest request)
         {
-            var product = _catalogLibrary.GetProduct(request.ProductSku);
+            Search.Models.Product product =
```

Comment:
Why the need to specify the namespace if we already have an alias on line 13?
Is there a chance to completely remove the `using` of  `EntitiesV2` (and thus the aliases as well) at this point?

Reply:
we don't need to specify namespaces, you are right.
But we can't remove EnititesV2 yet, we get the Currency from there (line 91)